src/components/application_manager/src/resumption/resumption_data_json.cc

```diff
@@ -68,7 +68,7 @@ void ResumptionDataJson::SaveApplication(
   const mobile_apis::HMILevel::eType hmi_level =
       application->hmi_level(mobile_apis::PredefinedWindows::DEFAULT_WINDOW);
   const bool is_subscribed_for_way_points =
-      application_manager_.IsAppSubscribedForWayPoints(application);
+      application_manager_.IsAppSubscribedForWayPoints(*application);
```

Comment:
I think, because during saving of resumption data reference, not pointer, to app is used, and sdl_app_extension also stores application in private field with reference to it.